src/main/java/org/threadly/litesockets/UDPServer.java

```diff
@@ -158,16 +155,15 @@ public void close(Throwable error) {
   }
 
   protected int doWrite() {
-    Pair<InetSocketAddress, ByteBuffer> wdp = writeQueue.poll();
-    SettableListenableFuture<Long> slf = writeFutures.poll();
-    if(wdp != null) {
+    WriteData wd = writeQueue.poll();
+    if(wd != null) {
       try {
-        return channel.send(wdp.getRight(), wdp.getLeft());
+        return channel.send(wd.getBuffer(), wd.getAddress());
       } catch (IOException e) {
         return 0;
       } finally {
-        if(slf != null) {
-          slf.setResult(0L);
+        if(wd.getSlf() != null && !wd.getSlf().isDone()) {
```

Comment:
Can use the ClientSettableListenableFuture since these writes technically dont have to go through the client.  They are call dont on the UDPServer and can be called with no UDPClient.  I do agree setting to false is a better way of dealing with it.